statannot/statannot.py

```diff
@@ -100,149 +104,88 @@ def stat_test(
-        result.test_str = result.test_str + ' with Bonferroni correction'
-    elif comparisons_correction is None:
-        pass
-    else:
-        # This should never be reached because `comparisons_correction` must
-        # be a valid correction method or None.
-        raise RuntimeError('Unexpectedly reached end of switch.')
+    # Optionally, run multiple comparisons correction that can separately be applied to each pval
+    if comparisons_correction is not None and comparisons_correction.type == 0:
+        result.pval = comparisons_correction(result.pval, num_comparisons)
+        result.correction_method = comparisons_correction.name
 
     return result
 
 
-def bonferroni(p_values, num_comparisons='auto'):
-    """Apply Bonferroni correction for multiple comparisons.
-
-    The Bonferroni correction is defined as
-        p_corrected = min(num_comparisons * p, 1.0).
-
-    Arguments
-    ---------
-    p_values: scalar or list-like
-        One or more p_values to correct.
-    num_comparisons: int or `auto`
-        Number of comparisons. Use `auto` to infer the number of comparisons
-        from the length of the `p_values` list.
-
-    Returns
-    -------
-    Scalar or numpy array of corrected p-values.
-
-    """
-    # Input checks.
-    if np.ndim(p_values) > 1:
-        raise_expected_got(
-            'Scalar or list-like', 'argument `p_values`', p_values
-        )
-    if num_comparisons != 'auto':
-        try:
-            # Raise a TypeError if num_comparisons is not numeric, and raise
-            # an AssertionError if it isn't int-like.
-            assert np.ceil(num_comparisons) == num_comparisons
-        except (AssertionError, TypeError) as e:
-            raise_expected_got(
-                'Int or `auto`', 'argument `num_comparisons`', num_comparisons
-            )
-
-    # Coerce p_values to numpy array.
-    p_values_array = np.atleast_1d(p_values)
-
-    if num_comparisons == 'auto':
-        # Infer number of comparisons
-        num_comparisons = len(p_values_array)
-    elif len(p_values_array) > 1 and num_comparisons != len(p_values_array):
-        # Warn if multiple p_values have been passed and num_comparisons is
-        # set manually.
-        warnings.warn(
-            'Manually-specified `num_comparisons={}` differs from number of '
-            'p_values to correct ({}).'.format(
-                num_comparisons, len(p_values_array)
-            )
-        )
-
-    # Apply correction by multiplying p_values and thresholding at p=1.0
-    p_values_array *= num_comparisons
-    p_values_array = np.min(
-        [p_values_array, np.ones_like(p_values_array)], axis=0
-    )
-
-    if len(p_values_array) == 1:
-        # Return a scalar if input was a scalar.
-        return p_values_array[0]
-    else:
-        return p_values_array
-
-
-
-def pval_annotation_text(x, pvalue_thresholds):
+def pval_annotation_text(result: Union[List[StatResult], StatResult], pvalue_thresholds):
```

Comment:
Type hinting! This guy is serious!

Reply:
I really do appreciate them. 
To match this repo coding style, I refrain myself and only use them where I feel they are almost necessary.